packages/grafana-data/src/types/data.ts

```diff
@@ -8,6 +8,12 @@ export enum LoadingState {
   Error = 'Error',
 }
 
+export interface QueryResultDataSourceMeta {
+  datasourceId: number;
+  datasourceName: string; // If we have the ID... do we need the name?!?!
+  // Explicit classes will extend with their custom metadata fields
```

Comment:
This is only added because I don't know a good way to load a datasource by ID -- am I missing something?

Reply:
dooh -- this is only about line 12, not 14